### PR TITLE
opencolorio: 2.0.2 -> 2.1.2

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

update package

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by rastertail — APPROVED** [review]

File: pkgs/development/libraries/opencolorio/default.nix [hunk 1]
```diff
@@ -1,40 +1,55 @@
...
+, GLUT
+, Cocoa
+# Python bindings
+, pythonBindings ? true # Python bindings
+, python3Packages
+# Build apps
+, buildApps ? true # Utility applications
+, lcms2
+, openimageio2
+, openexr }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/opencolorio/default.nix:24]:
```suggestion
, openexr
}:
```

**Comment by risicle** [comment]
This broke x86_64-darwin.

**Comment by onny** [comment]
> This broke x86_64-darwin.

Thank you for noticing. I'll investigate

**Comment by risicle** [comment]
The immediate failure I've fixed by adding a patch reverting https://github.com/AcademySoftwareFoundation/OpenColorIO/commit/48bab7c643ed8d108524d718e5038d836f906682 - presumably our darwin x86_64 macos SDK doesn't support this `-hidden-l` flag, but that reveals the larger problem that 2.1.2 wants to enable Metal support, which I assume our old SDK doesn't support.

**Comment by risicle** [comment]
And no they didn't think to add the option to disable Metal support.

**Comment by risicle** [comment]
Think I might have a fix...

**Pull request merged**